blockchain/indexers/cfindex.go

```diff
@@ -55,6 +66,19 @@ func dbFetchFilter(dbTx database.Tx, key []byte, h *chainhash.Hash) ([]byte, err
 	return idx.Get(h[:]), nil
 }
 
+// dbFetchFilterHash retrieves a block's basic or extended filter header.
+// A filter's absence is not considered an error.
+func dbFetchFilterHash(dbTx database.Tx, key []byte, h *chainhash.Hash) ([]byte, error) {
+	idx := dbTx.Metadata().Bucket(cfIndexParentBucketKey).Bucket(key)
+
+	fh := idx.Get(h[:])
+	if len(fh) != fastsha256.Size {
```

Comment:
Is `fastsha` still used in this package? Upstream `btcd` has abandoned it in favor of the standard lib recently.

Reply:
Shouldn't this be returning `sha256.sum256` of the filter itself? As is, looks like this will _always_ fail.

Reply:
Perhaps we should actually just manually hash the filter itself? Tradeoff between space/time I guess.

Reply:
1) I just copied the `dbFetchFilterHeader` method wholesale. I can replace `fastsha256` everywhere in this file if you want, but I'd do it in a separate commit.
2) I don't understand why this should fail if the key argument is correct, it should look up a hash. 
3) Yeah, it's a time/IO concern -- I don't think it should read 2,500 filters then hash them, when they are easily stored.